neo/rawio/intanrawio.py

```diff
@@ -114,8 +179,23 @@ def _parse_header(self):
             signal_streams["name"][stream_index] = stream_type_to_name.get(int(stream_id), "")
 
         self._max_sampling_rate = np.max(signal_channels["sampling_rate"])
-        self._max_sigs_length = self._raw_data.size * self._block_size
 
+        if self.file_format == 'header-attached':
+            self._max_sigs_length = self._raw_data.size * self._block_size
+        elif self.file_format == 'one-file-per-signal':
+             self._max_sigs_length = max(
+                [
+                    raw_data.size
+                    for raw_data in self._raw_data.values()
+                ]
+            )
+        else:
+             self._max_sigs_length = max(
+                [
+                    len(raw_data) * raw_data[0].size
+                    for raw_data in self._raw_data.values()
+                ]
+            )
```

Comment:
I was thinking about this? Maybe just the raw_data[0].size? since each is an individual file?

Reply:
Do we need this? Can you just calculate the `get_signal_size` over all the stream indices either at the end of parsing or within `_segment_t_stop`?

Reply:
This is based on Sam's implementation. It occurs here so I kept it here for everything.

Reply:
Yeah, I think he was just pre-computing this value to use it when `_segment_t_stop` is called. But we can discuss that in another PR.